### ISSUE TITLE
Unhandled exception in Magazine Search - bi-monthly magazine

### ISSUE THREAD

**Issue by sscraggles** [DobyTang/LazyLibrarian#933]
Force Search from Magazine page results in search terminating with unhandled exception error.  This particular mag has two month names in the filename to denote a bi-monthly magazine.

```
23-Jul-2017 03:02:28 - DEBUG   :: SEARCHALLMAG : Found 14 results from Limetorrent for Professional Motorsport World
23-Jul-2017 03:02:28 - DEBUG   :: SEARCHALLMAG : Magazine matched: 100% Professional Motorsport World for Professional Motorsport World TruePDF March April 2017
23-Jul-2017 03:02:28 - DEBUG   :: SEARCHALLMAG : Magazine Professional Motorsport World TruePDF March April 2017 incorrect date or issue format.
23-Jul-2017 03:02:28 - DEBUG   :: SEARCHALLMAG : Magazine matched: 100% Professional Motorsport World for Professional Motorsport World Expo Special TruePDF Issue 2016
23-Jul-2017 03:02:28 - ERROR   :: SEARCHALLMAG : Unhandled exception in search_magazines: Traceback (most recent call last):
  File "/opt/LazyLibrarian/lazylibrarian/searchmag.py", line 411, in search_magazines
    comp_date = int(newdatish) - int(control_date)
ValueError: invalid literal for int() with base 10: 'Professional.Motorsport.World.March.June.2'
```

### LazyLibrarian version number (at the bottom of config page)
21f1300855c402cef59e17864ebafb4b52c99076
### Operating system used (windows, mac, linux, NAS type)
Debian 9
### Interface in use (default, bookstrap)
Bookstrap
### Which api (Goodreads, GoogleBooks, both)
Goodreads
### Source of your LazyLibrarian installation (git, zip file, 3rd party package)
Git clone
### Relevant debug log with api keys and any passwords redacted

**Comment by sscraggles** [comment]
Maybe it's not the bi-monthly date, it's failing on others too....

```
23-Jul-2017 03:21:18 - DEBUG   :: SEARCHALLMAG : Magazine Top Gear UK The Perfect Road Trip 2013 v rar not in a recognised date format.
23-Jul-2017 03:21:18 - DEBUG   :: SEARCHALLMAG : Magazine Top Gear UK The Perfect Road Trip 2013 v rar incorrect date or issue format.
23-Jul-2017 03:21:18 - DEBUG   :: SEARCHALLMAG : Magazine matched: 100% Top Gear UK for Top Gear UK The Perfect Road Trip 2013 DVDRip x264 iOM rar
23-Jul-2017 03:21:18 - DEBUG   :: SEARCHALLMAG : Magazine Top Gear UK The Perfect Road Trip 2013 DVDRip x264 iOM rar not in a recognised date format.
23-Jul-2017 03:21:18 - DEBUG   :: SEARCHALLMAG : Magazine Top Gear UK The Perfect Road Trip 2013 DVDRip x264 iOM rar incorrect date or issue format.
23-Jul-2017 03:21:18 - DEBUG   :: SEARCHALLMAG : Magazine matched: 100% Top Gear UK for Top Gear UK The Perfect Road Trip 2013 v rar
23-Jul-2017 03:21:18 - DEBUG   :: SEARCHALLMAG : Magazine Top Gear UK The Perfect Road Trip 2013 v rar not in a recognised date format.
23-Jul-2017 03:21:18 - DEBUG   :: SEARCHALLMAG : Magazine Top Gear UK The Perfect Road Trip 2013 v rar incorrect date or issue format.
23-Jul-2017 03:21:18 - DEBUG   :: SEARCHALLMAG : Magazine matched: 100% Top Gear UK for Top Gear UK The Perfect Road Trip 2 2014 DVDRiP X264 TASTE
23-Jul-2017 03:21:18 - ERROR   :: SEARCHALLMAG : Unhandled exception in search_magazines: Traceback (most recent call last):
  File "/opt/LazyLibrarian/lazylibrarian/searchmag.py", line 411, in search_magazines
    comp_date = int(newdatish) - int(control_date)
ValueError: invalid literal for int() with base 10: 'September 2016'
```

**Comment by philborman** [comment]
Seems to be failing on magazine date parsing, thinks it's got an issue number rather than a date and then can't turn the number into an integer for comparison. The date parsing is quite lengthy, can you put debug level up to 4 and scan again, should get a line saying "regex nn [nzb download title]  something"
The nn is which regex it thinks it passed, which would narrow it down a lot. Thanks

**Comment by sscraggles** [comment]
Here's a log, using www-limetorrents-cc.proxydude.xyz

```
23-Jul-2017 12:34:54 - DEBUG   :: SEARCHALLMAG : Found 14 results from Limetorrent for Professional Motorsport World
23-Jul-2017 12:34:54 - DEBUG   :: SEARCHALLMAG : Magazine matched: 100% Professional Motorsport World for Professional Motorsport World TruePDF March April 2017
23-Jul-2017 12:34:54 - DEBUG   :: SEARCHALLMAG : regex 1 [Professional Motorsport World TruePDF March April 2017] 2017-04-01
23-Jul-2017 12:34:54 - DEBUG   :: SEARCHALLMAG : Magazine Professional Motorsport World TruePDF March April 2017 incorrect date or issue format.
23-Jul-2017 12:34:55 - DEBUG   :: SEARCHALLMAG : Added Professional.Motorsport.World.TruePDF-March.April.2017 to pastissues marked Skipped
23-Jul-2017 12:34:55 - DEBUG   :: SEARCHALLMAG : Magazine matched: 100% Professional Motorsport World for Professional Motorsport World Expo Special TruePDF Issue 2016
23-Jul-2017 12:34:55 - DEBUG   :: SEARCHALLMAG : regex 5 [Professional Motorsport World Expo Special TruePDF Issue 2016] 2016
23-Jul-2017 12:34:55 - ERROR   :: SEARCHALLMAG : Unhandled exception in search_magazines: Traceback (most recent call last):
  File "/opt/LazyLibrarian/lazylibrarian/searchmag.py", line 411, in search_magazines
    comp_date = int(newdatish) - int(control_date)
ValueError: invalid literal for int() with base 10: 'Professional.Motorsport.World.March.June.2'

```

**Comment by sscraggles** [comment]
another...
```
23-Jul-2017 12:40:29 - DEBUG   :: SEARCHALLMAG : Magazine matched: 100% Top Gear UK for Top Gear UK The Perfect Road Trip 2013 v rar
23-Jul-2017 12:40:29 - DEBUG   :: SEARCHALLMAG : Magazine Top Gear UK The Perfect Road Trip 2013 v rar not in a recognised date format.
23-Jul-2017 12:40:29 - DEBUG   :: SEARCHALLMAG : regex 99 [Top Gear UK The Perfect Road Trip 2013 v rar] 1970-01-01
23-Jul-2017 12:40:29 - DEBUG   :: SEARCHALLMAG : Magazine Top Gear UK The Perfect Road Trip 2013 v rar incorrect date or issue format.
23-Jul-2017 12:40:29 - DEBUG   :: SEARCHALLMAG : Top Gear UK The Perfect Road Trip 2013 v rar is already in pastissues marked Skipped
23-Jul-2017 12:40:29 - DEBUG   :: SEARCHALLMAG : Magazine matched: 100% Top Gear UK for Top Gear UK The Perfect Road Trip 2 2014 DVDRiP X264 TASTE
23-Jul-2017 12:40:29 - DEBUG   :: SEARCHALLMAG : regex 6 [Top Gear UK The Perfect Road Trip 2 2014 DVDRiP X264 TASTE] 0
23-Jul-2017 12:40:29 - ERROR   :: SEARCHALLMAG : Unhandled exception in search_magazines: Traceback (most recent call last):
  File "/opt/LazyLibrarian/lazylibrarian/searchmag.py", line 411, in search_magazines
    comp_date = int(newdatish) - int(control_date)
ValueError: invalid literal for int() with base 10: 'September 2016'

```

**Comment by sscraggles** [comment]
Disabled limetorrents...
```
23-Jul-2017 12:42:44 - DEBUG   :: SEARCHALLMAG : [IterateOverTorrentSites] - https://pirateproxy.cc
23-Jul-2017 12:42:46 - DEBUG   :: SEARCHALLMAG : Parsing results from <a href="https://pirateproxy.cc/s/?q=Professional+Motorsport+World&category=0&orderby=99&page=0">TPB</a>
23-Jul-2017 12:42:46 - DEBUG   :: SEARCHALLMAG : Found 0 results from TPB for Professional Motorsport World
23-Jul-2017 12:42:46 - DEBUG   :: SEARCHALLMAG : Magazine matched: 100% Professional Motorsport World for Professional Motorsport World TruePDF March April 2017
23-Jul-2017 12:42:46 - DEBUG   :: SEARCHALLMAG : regex 1 [Professional Motorsport World TruePDF March April 2017] 2017-04-01
23-Jul-2017 12:42:46 - DEBUG   :: SEARCHALLMAG : Magazine Professional Motorsport World TruePDF March April 2017 incorrect date or issue format.
23-Jul-2017 12:42:46 - DEBUG   :: SEARCHALLMAG : Added Professional.Motorsport.World.TruePDF-March.April.2017 to pastissues marked Skipped
23-Jul-2017 12:42:46 - DEBUG   :: SEARCHALLMAG : Magazine matched: 100% Professional Motorsport World for Professional Motorsport World Expo Special TruePDF Issue 2016
23-Jul-2017 12:42:46 - DEBUG   :: SEARCHALLMAG : regex 5 [Professional Motorsport World Expo Special TruePDF Issue 2016] 2016
23-Jul-2017 12:42:46 - ERROR   :: SEARCHALLMAG : Unhandled exception in search_magazines: Traceback (most recent call last):
  File "/opt/LazyLibrarian/lazylibrarian/searchmag.py", line 411, in search_magazines
    comp_date = int(newdatish) - int(control_date)
ValueError: invalid literal for int() with base 10: 'Professional.Motorsport.World.March.June.2'
```

**Comment by philborman** [comment]
ok, I think I see the problem. 
We are getting mixes of issue numbers and dates for the same magazine and can't handle it. We only store either latest issue number, or latest issue date, and assume all other issues of that magazine be the same. I have added some extra debugging messages and code to stop the ValueError, but not sure how to handle it for real. For now we will complain and put the issue in the past-issues list.
Example:
 Modern Mini Issue 85 TruePDF Spring 2017  we can pick up as either issue or date. 
If we store it as Issue 85, next one should be Issue 86, but if it says April 2017 is that earlier or later.
If we store it as Spring 2017 the next one should be Summer 2017, but might say Issue 86.
It's a bit more complicated than that as we convert Spring 2017 to 2017-04-01, but you see the problem.

**Comment by sscraggles** [comment]
Yes, Professional Motorsport World is failing because as well as a Monthly there is an annual named "Professional Motorsport World Expo Special TruePDF Issue 2016"
Similarly for the monthly "Top Gear UK" magazine matching a TV episode of  "Top Gear UK The Perfect Road Trip 2 2014"  or "Top Gear UK The Perfect Road Trip 2013"
Modern Mini Spring, I guess is probably best ignored as Scene naming varies and can also be put out as "March-April 2017" or "May-June 2017".

I would likely tidy up in Calibre, though that's another foible as I have multiple Calibre libraries so Lazylibrarian doesn't pick them up.

**Comment by philborman** [comment]
Right, so if we hit the annual one first, we think the magazine is based around issue numbers, latest being 2016, and the ones with a date fail. Easy way around it is to put two magazines into lazylibrarian, the regular monthly should have an exclude keyword of "Expo" or "Special".  The second should have a title of "Professional Motorsport World Expo Special". That way the monthly is date based and the annual is issue numbers.  we should already be able to handle two-month issues, we just read the month name nearest to the year so "May-June 2017" reads as June, "2017 May-June" would read as May.

**Comment by philborman** [comment]
Just pushed a new version with error trapping for the integer issue, and more messages. Should help.

**Comment by sscraggles** [comment]
Updated to 12b3f15f84000fb751a336842b9e19bf84b49ef7
Got another unhandled exception on the Spring title
```
23-Jul-2017 17:23:43 - DEBUG   :: SEARCHALLMAG : [NewzNabPlus] - result fields from NZB are {'nzbmode': 'nzb', 'nzbdate': 'Thu, 07 May 2015 06:48:07 +0000', 'bookid': u'Modern Mini', 'nzbtitle': 'Modern.Mini-Summer.2015', 'nzbsize': '51655000', 'nzburl': 'https://api.nzbgeek.info/api?t=get&id=fa46a83c74575e4d089e93da1fe0a821&apikey=mykey', 'nzbprov': u'https://api.nzbgeek.info', 'priority': 0}
23-Jul-2017 17:23:43 - DEBUG   :: SEARCHALLMAG : [NewzNabPlus] - result fields from NZB are {'nzbmode': 'nzb', 'nzbdate': 'Fri, 16 Jan 2015 02:32:45 +0000', 'bookid': u'Modern Mini', 'nzbtitle': 'Modern.Mini-March.April.2015', 'nzbsize': '53806000', 'nzburl': 'https://api.nzbgeek.info/api?t=get&id=567990aa885ab49f5cdf611062100838&apikey=mykey', 'nzbprov': u'https://api.nzbgeek.info', 'priority': 0}
23-Jul-2017 17:23:43 - DEBUG   :: SEARCHALLMAG : [NewzNabPlus] - result fields from NZB are {'nzbmode': 'nzb', 'nzbdate': 'Fri, 05 Dec 2014 06:56:59 +0000', 'bookid': u'Modern Mini', 'nzbtitle': 'Modern.Mini-January.February.2015', 'nzbsize': '43468000', 'nzburl': 'https://api.nzbgeek.info/api?t=get&id=e753cd6f6d4b7f5fbf7cce096aa43dec&apikey=mykey', 'nzbprov': u'https://api.nzbgeek.info', 'priority': 0}
23-Jul-2017 17:23:43 - DEBUG   :: SEARCHALLMAG : [NewzNabPlus] - result fields from NZB are {'nzbmode': 'nzb', 'nzbdate': 'Tue, 18 Nov 2014 20:02:48 +0000', 'bookid': u'Modern Mini', 'nzbtitle': 'Modern.Mini.TruePDF-January.February.2015', 'nzbsize': '43455000', 'nzburl': 'https://api.nzbgeek.info/api?t=get&id=8d8b7ac067078ad81c3c430e13876445&apikey=mykey', 'nzbprov': u'https://api.nzbgeek.info', 'priority': 0}
23-Jul-2017 17:23:43 - DEBUG   :: SEARCHALLMAG : [NewzNabPlus] - result fields from NZB are {'nzbmode': 'nzb', 'nzbdate': 'Tue, 07 Oct 2014 09:13:11 +0000', 'bookid': u'Modern Mini', 'nzbtitle': 'Modern.Mini.TruePDF-November.December.2014', 'nzbsize': '41807000', 'nzburl': 'https://api.nzbgeek.info/api?t=get&id=167692f82c42589a54626db352e8fe82&apikey=mykey', 'nzbprov': u'https://api.nzbgeek.info', 'priority': 0}
23-Jul-2017 17:23:43 - DEBUG   :: SEARCHALLMAG : [NewzNabPlus] - result fields from NZB are {'nzbmode': 'nzb', 'nzbdate': 'Wed, 17 Sep 2014 13:52:01 +0000', 'bookid': u'Modern Mini', 'nzbtitle': 'Modern.Mini-September.October.2014', 'nzbsize': '41226000', 'nzburl': 'https://api.nzbgeek.info/api?t=get&id=1a60db0c8b4ce3cff3554aa8e18832f0&apikey=mykey', 'nzbprov': u'https://api.nzbgeek.info', 'priority': 0}
23-Jul-2017 17:23:43 - DEBUG   :: SEARCHALLMAG : [NewzNabPlus] - result fields from NZB are {'nzbmode': 'nzb', 'nzbdate': 'Thu, 27 Mar 2014 20:22:07 +0000', 'bookid': u'Modern Mini', 'nzbtitle': 'Modern.Mini-May.June.2014', 'nzbsize': '47848000', 'nzburl': 'https://api.nzbgeek.info/api?t=get&id=c2e59b80584902fe023fa5c71554cf5a&apikey=mykey', 'nzbprov': u'https://api.nzbgeek.info', 'priority': 0}
23-Jul-2017 17:23:43 - DEBUG   :: SEARCHALLMAG : [NewzNabPlus] - result fields from NZB are {'nzbmode': 'nzb', 'nzbdate': 'Tue, 25 Mar 2014 23:29:33 +0000', 'bookid': u'Modern Mini', 'nzbtitle': 'Modern.Mini.Magazine.TruePDF-May.June.2014', 'nzbsize': '47862000', 'nzburl': 'https://api.nzbgeek.info/api?t=get&id=b8597e347e8d6ef99fe6e7ca2e2d9252&apikey=mykey', 'nzbprov': u'https://api.nzbgeek.info', 'priority': 0}
23-Jul-2017 17:23:43 - DEBUG   :: SEARCHALLMAG : Found 26 nzb at https://api.nzbgeek.info for: Modern Mini
23-Jul-2017 17:23:43 - DEBUG   :: SEARCHALLMAG : [IterateOverTorrentSites] - https://kickass-cd.proxydude.xyz/
23-Jul-2017 17:23:45 - DEBUG   :: SEARCHALLMAG : Parsing results from <a href="https://kickass-cd.proxydude.xyz//usearch/Modern%20Mini/?category=books&field=seeders&sorder=desc">KAT</a>
23-Jul-2017 17:23:45 - DEBUG   :: SEARCHALLMAG : Found 0 results from KAT for Modern Mini
23-Jul-2017 17:23:45 - DEBUG   :: SEARCHALLMAG : [IterateOverTorrentSites] - https://pirateproxy.cc
23-Jul-2017 17:23:46 - DEBUG   :: SEARCHALLMAG : Parsing results from <a href="https://pirateproxy.cc/s/?q=Modern+Mini&category=0&orderby=99&page=0">TPB</a>
23-Jul-2017 17:23:46 - DEBUG   :: SEARCHALLMAG : Found 0 results from TPB for Modern Mini
23-Jul-2017 17:23:46 - DEBUG   :: SEARCHALLMAG : [IterateOverTorrentSites] - https://www-limetorrents-cc.proxydude.xyz
23-Jul-2017 17:23:47 - DEBUG   :: SEARCHALLMAG : Parsing results from <a href="https://www-limetorrents-cc.proxydude.xyz/searchrss/other?q=Modern+Mini">Limetorrent</a>
23-Jul-2017 17:23:47 - DEBUG   :: SEARCHALLMAG : Found Modern Mini - May, June 2015 {bk} but 1 seeder
23-Jul-2017 17:23:47 - DEBUG   :: SEARCHALLMAG : Found Last Demon (Penguin Mini Modern Classics)  epub but 2 seeders
23-Jul-2017 17:23:47 - DEBUG   :: SEARCHALLMAG : Found Fruupp - Modern Masquerades (2016) Mini LP SHM Blu-spec FLAC Beolab1700 but 0 seeders
23-Jul-2017 17:23:47 - DEBUG   :: SEARCHALLMAG : Found 0 results from Limetorrent for Modern Mini
23-Jul-2017 17:23:47 - DEBUG   :: SEARCHALLMAG : Magazine matched: 100% Modern Mini for Modern Mini Issue 85 TruePDF Spring 2017
23-Jul-2017 17:23:47 - DEBUG   :: SEARCHALLMAG : regex 1 [Modern Mini Issue 85 TruePDF Spring 2017] 2017-04-01
23-Jul-2017 17:23:47 - ERROR   :: SEARCHALLMAG : Unhandled exception in search_magazines: Traceback (most recent call last):
  File "/opt/LazyLibrarian/lazylibrarian/searchmag.py", line 412, in search_magazines
    comp_date = datecompare(newdatish, control_date)
  File "/opt/LazyLibrarian/lazylibrarian/formatter.py", line 186, in datecompare
    y2 = int(control_date.split('-')[0])
ValueError: invalid literal for int() with base 10: 'Modern Mini 2017 May'
```

**Comment by sscraggles** [comment]
Also seeing it download the annuals, despite having reject filters set as "Special, Expo"
```
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : Magazine matched: 100% Professional Motorsport World for Professional Motorsport World TruePDF March April 2017
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : regex 1 [Professional Motorsport World TruePDF March April 2017] 2017-04-01
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : Magazine date comparing to 2016-07-23
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : This issue of Professional Motorsport World TruePDF March April 2017 is new, downloading
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : Magazine request number 1
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : [u'Professional Motorsport World,2017-04-01']
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : Added Professional.Motorsport.World.TruePDF-March.April.2017 to wanted marked Wanted
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : Magazine matched: 100% Professional Motorsport World for Professional Motorsport World Expo Special TruePDF Issue 2016
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : regex 5 [Professional Motorsport World Expo Special TruePDF Issue 2016] 2016
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : Magazine comparing issue numbers (2016)
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : This issue of Professional Motorsport World Expo Special TruePDF Issue 2016 is new, downloading
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : Magazine request number 2
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : [u'Professional Motorsport World,2017-04-01', u'Professional Motorsport World,2016']
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : Added Professional.Motorsport.World.Expo.Special.TruePDF-Issue.2016 to wanted marked Wanted
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : Magazine matched: 100% Professional Motorsport World for [MAGAZINE] Professional Motorsport World TruePDF July August 2016 pdf
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : regex 1 [[MAGAZINE] Professional Motorsport World TruePDF July August 2016 pdf] 2016-08-01
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : Magazine date comparing to 2016-07-23
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : This issue of [MAGAZINE] Professional Motorsport World TruePDF July August 2016 pdf is new, downloading
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : Magazine request number 3
23-Jul-2017 17:50:25 - DEBUG   :: SEARCHMAG : [u'Professional Motorsport World,2017-04-01', u'Professional Motorsport World,2016', u'Professional Motorsport World,2016-08-01']

```

**Comment by philborman** [comment]
Just added some more debugging lines to try and track this down. Reject word list should be actioned before we get to date parsing. I'm also not sure why we are sometimes calling datecompare with strings that aren't dates, eg 
 File "/opt/LazyLibrarian/lazylibrarian/searchmag.py", line 412, in search_magazines
    comp_date = datecompare(newdatish, control_date)
  File "/opt/LazyLibrarian/lazylibrarian/formatter.py", line 186, in datecompare
    y2 = int(control_date.split('-')[0])
ValueError: invalid literal for int() with base 10: 'Modern Mini 2017 May'

We should only call the datecompare routine if both newdatish and control_date are in the format yyyy-dd-mm  I have added a more severe check for this.

It might be best to remove the old magazines from your library and rescan as we might have incorrect values stored in "latest issue" column

**Comment by sscraggles** [comment]
Started up a clean test instance, added magazines:
Professional Motorsport World (exclude filters: Special, Expo)
3 torrent providers and 1 newssnab

```
26-Jul-2017 17:37:32 - DEBUG   :: SEARCHMAG : Magazine matched: 100% Professional Motorsport World for Professional Motorsport World TruePDF March April 2017
26-Jul-2017 17:37:32 - DEBUG   :: SEARCHMAG : Reject: ['Special', 'Expo']
26-Jul-2017 17:37:32 - DEBUG   :: SEARCHMAG : Title: professional motorsport world truepdf march april 2017
26-Jul-2017 17:37:32 - DEBUG   :: SEARCHMAG : Bookid: professional motorsport world
26-Jul-2017 17:37:32 - DEBUG   :: SEARCHMAG : regex 1 [Professional Motorsport World TruePDF March April 2017] 2017-04-01
26-Jul-2017 17:37:32 - DEBUG   :: SEARCHMAG : Magazine date comparing to 2016-07-26
26-Jul-2017 17:37:32 - DEBUG   :: SEARCHMAG : This issue of Professional Motorsport World TruePDF March April 2017 is new, downloading
26-Jul-2017 17:37:32 - DEBUG   :: SEARCHMAG : Magazine request number 1
26-Jul-2017 17:37:32 - DEBUG   :: SEARCHMAG : [u'Professional Motorsport World,2017-04-01']
26-Jul-2017 17:37:32 - DEBUG   :: SEARCHMAG : Added Professional.Motorsport.World.TruePDF-March.April.2017 to wanted marked Wanted
26-Jul-2017 17:37:32 - DEBUG   :: SEARCHMAG : Magazine matched: 100% Professional Motorsport World for Professional Motorsport World Expo Special TruePDF Issue 2016
26-Jul-2017 17:37:32 - DEBUG   :: SEARCHMAG : Reject: ['Special', 'Expo']
26-Jul-2017 17:37:32 - DEBUG   :: SEARCHMAG : Title: professional motorsport world expo special truepdf issue 2016
26-Jul-2017 17:37:32 - DEBUG   :: SEARCHMAG : Bookid: professional motorsport world
26-Jul-2017 17:37:32 - DEBUG   :: SEARCHMAG : regex 5 [Professional Motorsport World Expo Special TruePDF Issue 2016] 2016
26-Jul-2017 17:37:32 - DEBUG   :: SEARCHMAG : Magazine comparing issue numbers (2016)
26-Jul-2017 17:37:32 - ERROR   :: SEARCHMAG : Unhandled exception in search_magazines: Traceback (most recent call last):
  File "/opt/LazyLibrarian.test/lazylibrarian/searchmag.py", line 414, in search_magazines
    if re.match('\d+-\d\d-\d\d', control_date) and re.match('\d+-\d\d-\d\d', newdatish):
  File "/usr/lib/python2.7/re.py", line 141, in match
    return _compile(pattern, flags).match(string)
TypeError: expected string or buffer
```

Disabled all providers and tried individually, and can only replicate when newsnab provider is enabled.

**Comment by philborman** [comment]
That's an easy fix. We think we're comparing an issue number (2016) but expecting a string.
Still not rejecting on the keywords either, case comparison issue. I think if you put your reject list in lowercase it will work. 
Just working on some other bits, will push a fix in an hour or two.

**Comment by sscraggles** [comment]
Tried again with 85013f82f88d9b978ac746e3f92da1d3aa292c5e

Same error with or without reject filter set.

```
30-Jul-2017 01:18:46 - DEBUG   :: SEARCHALLMAG : Magazine matched: 100% Professional Motorsport World for Professional Motorsport World Expo Special TruePDF Issue 2016
30-Jul-2017 01:18:46 - DEBUG   :: SEARCHALLMAG : Reject: ['Expo', 'Special']
30-Jul-2017 01:18:46 - DEBUG   :: SEARCHALLMAG : Title: professional motorsport world expo special truepdf issue 2016
30-Jul-2017 01:18:46 - DEBUG   :: SEARCHALLMAG : Bookid: professional motorsport world
30-Jul-2017 01:18:46 - DEBUG   :: SEARCHALLMAG : regex 5 [Professional Motorsport World Expo Special TruePDF Issue 2016] 2016
30-Jul-2017 01:18:46 - DEBUG   :: SEARCHALLMAG : Magazine comparing issue numbers (2016)
30-Jul-2017 01:18:46 - ERROR   :: SEARCHALLMAG : Unhandled exception in search_magazines: Traceback (most recent call last):
  File "/opt/LazyLibrarian.test/lazylibrarian/searchmag.py", line 414, in search_magazines
    if re.match('\d+-\d\d-\d\d', control_date) and re.match('\d+-\d\d-\d\d', newdatish):
  File "/usr/lib/python2.7/re.py", line 141, in match
    return _compile(pattern, flags).match(string)
TypeError: expected string or buffer


```

**Comment by philborman** [comment]
Was fixed in 	a6ac5f6
I think you are not on current version

**Comment by sscraggles** [comment]
Ah yes, you're right.  I had edited one of the application files which was preventing the auto update.
No longer seeing any of these issues.
